community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPage.java

```diff
@@ -326,7 +326,7 @@ public int swapIn( StoreChannel channel, long offset, int length ) throws IOExce
             while ( read != -1 && (readTotal += read) < length );
 
             // Zero-fill the rest.
-            UnsafeUtil.setMemory( pointer + readTotal, getCachePageSize() - readTotal, (byte) 0 );
+            UnsafeUtil.setMemory( pointer + readTotal, length - readTotal, (byte) 0 );
```

Comment:
Correct. The `length` is the file-page-size, which is less than or equal to the cache page size.